src/presentation/device.rs

```diff
@@ -267,21 +370,51 @@ impl SessionManager {
         Ok(request)
     }
 
-    /// Handle a request from the reader.
+    /// Handle a new request from the reader.
+    ///
+    /// The request is expected to be a [CBOR](https://cbor.io)
+    /// encoded [SessionData] and encrypted.
+    /// It will parse and validate it.
+    ///
+    /// It returns the requested items by the reader.
     pub fn handle_request(&mut self, request: &[u8]) -> anyhow::Result<RequestedItems> {
         let session_data: SessionData = serde_cbor::from_slice(request)?;
         self.handle_decoded_request(session_data)
     }
 
-    /// Get the next payload for signing.
+    /// When there are documents to be signed, it will return then next one for signing.
+    ///
+    /// After signed, you need to call [SessionManager::submit_next_signature].
+    ///
+    /// # Example
+    ///
+    /// ```ignore
+    /// while let Some((_, payload)) = session_manager.get_next_signature_payload()? {
+    ///     let signature = sign(&payload);
+    ///     session_manager.submit_next_signature(signature);
+    /// }
+    /// ```
     pub fn get_next_signature_payload(&self) -> Option<(Uuid, &[u8])> {
         match &self.state {
             State::Signing(p) => p.get_next_signature_payload(),
```

Comment:
I don't believe this method is returning the `next` signature payload. It appears to only be returning the `last` signature payload.

A typical `next` method should consume the item, if it exists, mutating the the list by removing the item from the list.

This should likely be fixed in a follow up PR, addressing the implementation versus the documentation comments.

Reply:
https://github.com/radumarias/isomdl/blob/41823af175b5bb8f36f1f037b7750dd05c3960fa/src/presentation/device.rs#L362

Reply:
Looks like I may have spoke too soon-- `submit_next_signature` appears to `pop` the prepared document.

Reply:
yes, it's like a `Deque`.